runtime/vm/createramclass.cpp

```diff
@@ -2405,7 +2405,7 @@ internalCreateRAMClassFromROMClassImpl(J9VMThread *vmThread, J9ClassLoader *clas
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].prefixSize = 0;
 			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignment = OMR_MAX(sizeof(J9Class *), sizeof(UDATA));
-			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignedSize = sizeof(J9FlattenedClassCache) * (flattenedClassCache[0].offset > 0 ? flattenedClassCache[0].offset + 1 : 0);
+			allocationRequests[RAM_CLASS_FLATTENED_CLASS_CACHE].alignedSize = sizeof(J9FlattenedClassCache) * (flattenedClassCache[0].offset + 1);
```

Comment:
Is this correct now that there's a header?

(also, the math here is very confusing - could use some documentation)

Reply:
The dependency between the cache entry struct and header struct is a bad idea. Better to use a union in the cache struct.

Reply:
@gacholio @tajila Should I go to modify the definition of J9FlattenedClassCache, and use a union for the J9Class* clazz entry?

Reply:
The existing `J9FlattenedClassCache` should be renamed as `J9FlattenedClassCacheEntry`. 

Then define `union J9FlattenedClassCache { J9FlattenedClassCacheEntry ...; J9FlattenedClassCacheHeader ...}